pymrio/tools/iobalance.py

```diff
@@ -0,0 +1,189 @@
+""""
+
+A function for checking the balance of an IO system.
+
+The balance is checked by:
+1- Making sure that the matrix shapes are compatible 
+2- Checks if total inputs = total outputs
+3- Checks sector/product balance (sector/product inputs must = outputs)
+
+"""
+
+#%%
+#import packages
+
+import pymrio
+import pandas as pd
+import numpy as np
+
+
+#%%
+
+def system_is_balanced(Inter_industry: pd.DataFrame, value_added: pd.DataFrame, final_demand: pd.DataFrame, total_output:pd.DataFrame):
+    """Checks if the entire system is balanced by:
+    
+    1) First checks the shapes and makes sure they are correct
+    2) Then checks if the system is balanced
+    3) The checks if sector by sector is balanced
+    
+    Note: It has a one percent tolerance for balance
+
+    Parameters
+    ----------
+        Inter_industry: (pd.DataFrame)
+            InterIndustry exchange matrix (Z)
+        value_added: (pd.DataFrame) 
+            Value Added vector (VA)
+        final_demand: (pd.DataFrame)
+            Final demand (Y)
+        total_output: (pd.DataFrame)
+            Total Output Vector (X)
+    
+    Returns:
+        True: If all conditions are met
+        Error: If at least one condition is not met
+                  
+    """
+    
+    #Error message for wrong matrix shape  
+    class MatrixNotCompatibleError(Exception):
+        def __init__(self, message="Matrix/Vector shape is not compatible with InterIndustry Matrix"):
+            self.message = message
+            super().__init__(self.message)
+
+    #error message for unbalanced system
+    class SystemNotBalancedError(Exception):
```

Comment:
great idea of these, but please put all the classes defintion out of the function